### PR TITLE
Fix Enumerable#zip

### PR DESCRIPTION
**Pull request by kachick**

### CODE REVIEW

File: opal/corelib/helpers.rb [hunk 1]
```diff
@@ -11,18 +11,22 @@ def self.type_error(object, type, method = nil, coerced = nil)
     end
   end
 
-  def self.coerce_to(object, type, method)
+  def self.coerce_to(object, type, method, argument = undefined)
```
Comment (elia) [inline opal/corelib/helpers.rb]: Although there's probably no real case for this I think at this point this one should be `args` and used as `object.__send__ method, *args`

like in:

```
obj.to_enum(method = :each, *args) → enum
obj.to_enum(method = :each, *args) {|*args| block} → enum
```

File: opal/corelib/array.rb [hunk 2]
```diff
@@ -2300,7 +2300,7 @@ def zip(*others, &block)
         }
         others[j] = #{(
           Opal.coerce_to?(`o`, Array, :to_ary) ||
-          Opal.coerce_to!(`o`, Enumerator, :each)
+          Opal.coerce_to!(`o`, Enumerator, :to_enum, :each)
```
Comment (elia) [inline opal/corelib/array.rb]: I'm not sure why this can't be just `Opal.coerce_to!(`o`, Enumerator, :to_enum)`, can you explain?

Comment (kachick) [reply opal/corelib/array.rb]: `ruby/spec` checks the given argument as https://github.com/ruby/spec/blob/e489657352f3b9ec3b64a4d20160bd12f336bd3e/core/enumerable/fixtures/classes.rb#L145-L149 🤔

File: opal/corelib/helpers.rb (hunk 1 above)
Comment (kachick) [reply opal/corelib/helpers.rb]: It's right! Updated as https://github.com/opal/opal/pull/1986/commits/dcdfbb28503c6bad6b341c968abd4bbe725dac66

File: opal/corelib/array.rb (hunk 2 above)
Comment (elia) [reply opal/corelib/array.rb]: Ok, that's fine, passing an enumerator to `Array#zip` should be pretty rare anyway

**Comment by elia** [comment]
Can you please rebase so that CI can re-run? 🙏

**Comment by kachick** [comment]
Sure. Rebased!

**Comment by elia** [comment]
Thanks!

**Comment by kachick** [comment]
Thank you!

**Pull request merged**